cwms-data-api/src/main/java/cwms/cda/api/TextTimeSeriesController.java

```diff
@@ -0,0 +1,380 @@
+        TimeSeriesTextMode mode = ctx.queryParamAsClass(MODE, TimeSeriesTextMode.class).getOrDefault(TimeSeriesTextMode.ALL);
+
+        if(office == null ){
+           throw new IllegalArgumentException(OFFICE + " is a required parameter");
+        }
+
+        if(tsId == null ){
+            throw new IllegalArgumentException(NAME + " is a required parameter");
+        }
+
+        if (beginZdt == null) {
+            throw new IllegalArgumentException(BEGIN + " is a required parameter");
+        }
+
+
+        String formatHeader = ctx.header(Header.ACCEPT);
+        ContentType contentType = Formats.parseHeaderAndQueryParm(formatHeader, "");
+        try (Timer.Context ignored = markAndTime(GET_ALL)) {
+            DSLContext dsl = getDslContext(ctx);
+            TimeSeriesTextDao dao = getDao(dsl);
+
+            String textMask = "*";
+
+            TextTimeSeries textTimeSeries = dao.retrieveFromDao(mode, office, tsId, textMask,
+                    beginZdt, endZdt, versionZdt,
+                    maxVersion, minAttr, maxAttr);
+
+            ctx.contentType(contentType.toString());
+
+            String result = Formats.format(contentType, textTimeSeries);
+            ctx.result(result);
+
+            ctx.status(HttpServletResponse.SC_OK);
+        } catch (Exception ex) {
+            CdaError re =
+                    new CdaError("Failed to process request: " + ex.getLocalizedMessage());
+            logger.log(Level.SEVERE, re.toString(), ex);
+            ctx.status(HttpServletResponse.SC_INTERNAL_SERVER_ERROR).json(re);
+        }
+
+    }
+
+    @Nullable
+    private static ZonedDateTime queryParamAsZdt(Context ctx, String param, String timezone) {
+        ZonedDateTime beginZdt = null;
+        String begin = ctx.queryParam(param);
+        if (begin != null) {
+            beginZdt = DateUtils.parseUserDate(begin, timezone);
+        }
+        return beginZdt;
+    }
+
+    @Nullable
+    private static ZonedDateTime queryParamAsZdt(Context ctx, String param) {
+        return queryParamAsZdt(ctx, param, ctx.queryParamAsClass(TIMEZONE, String.class).getOrDefault("UTC"));
+    }
+
+    @OpenApi(ignore = true)
+    @Override
+    public void getOne(@NotNull Context ctx, @NotNull String templateId) {
+        throw new UnsupportedOperationException(NOT_SUPPORTED_YET);
+    }
+
+    @OpenApi(
+        description = "Create new TextTimeSeries",
+        requestBody = @OpenApiRequestBody(
+            content = {
+                @OpenApiContent(from = TextTimeSeries.class, type = Formats.JSONV2)
+            },
+            required = true),
+        queryParams = {
+
+            @OpenApiParam(name = REPLACE_ALL, type = Boolean.class, description = "Whether to replace any and all existing text with the specified text. Default is " + DEFAULT_CREATE_REPLACE_ALL)
+                        },
+        method = HttpMethod.POST,
+        tags = {TAG}
+    )
+    @Override
+    public void create(@NotNull Context ctx) {
```

Comment:
@MikeNeilson given the LRTS identifier updates coming down the pipeline. It it acceptable to not support storage of new LRTS text time series? Otherwise we'd need a- flag "store-as-lrts" similar to numerical time series.

I'd think skipping a "store-as-lrts" in both text and binary time series makes sense given we'll get the feature for "free" once the schema update is complete. This would mean we could turn on sessioning flags to require the new identifier and return the new identifier.